spec/controllers/api/v2/accounts/report_controller_spec.rb

```diff
@@ -223,6 +223,39 @@
         expect(response).to have_http_status(:success)
       end
     end
+
+    context 'when an agent has access to multiple accounts' do
+      let(:account1) { create(:account) }
+      let(:account2) { create(:account) }
+
+      let(:params) do
+        super().merge(
+          type: :agent,
+          since: 30.days.ago.to_i.to_s,
+          until: date_timestamp.to_s
+        )
+      end
+
+      it 'returns agent metrics from the current account' do
+        admin1 = create(:user, account: account1, role: :administrator)
```

Comment:
we haven't added admin1 as a member of account 2 here I think. 
we have to add
```
create(:account_user, user: admin1, account: account2)
```

Reply:
Done